lib/ansible/modules/cloud/misc/proxmox.py

```diff
@@ -347,7 +347,7 @@ def create_instance(module, proxmox, vmid, node, disk, storage, cpus, memory, sw
             kwargs.update(kwargs['mounts'])
             del kwargs['mounts']
         if 'pubkey' in kwargs:
-            if float(proxmox.version.get()['version']) >= 4.2:
+            if float(proxmox.version.get()['version'].split('-')[0]) >= 4.2:
```

Comment:
Have a look at #59356. You can find there a variable mapping.

Reply:
What do you mean by "variable mapping"? Here we are checking for version 4.2, if I convert the type to int, we can get undefined behavior when working with proxmox 4.1, for example

Reply:
I have the same problem, but this workaround doesn't helps me.